tests/utilities/test_io.py

```diff
@@ -0,0 +1,65 @@
+import os
+import stat
+from pathlib import Path
+from tempfile import mkdtemp, mkstemp
+
+import fsspec
+import pytest
+
+from seagap.utilities.io import check_file_exists, check_permission
+
+PREFIX = "seagap-"
+
+
+@pytest.fixture(params=["single", "glob", "random"])
+def input_path(request):
+    # TODO: Add remote file test
+    _, tmp_file = mkstemp(prefix=PREFIX)
+    file_path = Path(tmp_file)
+    file_dir = file_path.parent
+    if request.param == "single":
+        return file_path, str(file_path)
+    elif request.param == "glob":
+        # Get file directory 2 level above
+        dir_path = file_dir.parent / "**" / file_path.name
+        return file_path, str(dir_path)
+    elif request.param == "random":
+        return "./non_existent"
+
+
+def test_check_file_exists(input_path):
+    """Tests the `check_file_exists` io function"""
+    file_path = None
+    if isinstance(input_path, str):
+        # A non existent file ... only path string
+        expected_value = False
+        test_path = input_path
+    else:
+        expected_value = True
+        file_path, test_path = input_path
+
+    assert isinstance(check_file_exists(test_path), bool) is True
+    assert check_file_exists(test_path) is expected_value
+
+    # Clean up after
+    if file_path is not None:
```

Comment:
Looks like the recommended way of cleaning up fixtures is either `yield` or explicit finalizers: 
https://docs.pytest.org/en/6.2.x/fixture.html

Using either of these two mechanisms would remove the need return the file_path as part of the fixture.

Reply:
From my understanding of the link you sent me is that this `yield` is for fixture. This section of code is not a fixture. When I did try to change it to use yield I get the following 🤔 

```python
def test_check_file_exists(input_path):
    """Tests the `check_file_exists` io function"""
    file_path = None
    if isinstance(input_path, str):
        # A non existent file ... only path string
        expected_value = False
        test_path = input_path
    else:
        expected_value = True
        file_path, test_path = input_path

    assert isinstance(check_file_exists(test_path), bool) is True
    assert check_file_exists(test_path) is expected_value

    # Clean up after
    yield file_path
    os.unlink(file_path)
```

test result

```bash
tests/utilities/test_io.py::test_check_file_exists XFAIL ([NOTRUN] yield tests were removed in pytest 4.0 - test_check_file_exists will be ignored) 
```

The line above just checks if file exists, if so, then clean it up.

Reply:
Nvm... I think I finally figured out how to use this, and have implemented it in the code now. Thanks again for this suggestion! 😄